src/main.rs

```diff
@@ -96,3 +96,169 @@ fn main() {
+                name: inbound_packet_processor,
+                dependencies: [messenger, player_state, block_state, patchwork_state]
+            ),
+            (
+                module: services::connection::start,
+                name: connection_service,
+                dependencies: [messenger, player_state, patchwork_state, inbound_packet_processor]
+            ),
+            (
+                module: services::keep_alive::start,
+                name: keep_alive,
+                dependencies: [messenger]
+            )
+        );
+        trace!("Services Started");
+
+        // Temporary, maybe set them through CLI
+        std::env::set_var("PORT", "8600");
+        std::env::set_var("PEER_PORT", "8601");
+
+        let address = String::from("127.0.0.1");
+        let port = std::env::var("PORT").unwrap().parse::<u16>().unwrap();
+        let peer_port = std::env::var("PEER_PORT").unwrap().parse::<u16>().unwrap();
+
+        patchwork_state.sender().new_map(models::map::Peer {
+            port: peer_port.clone(),
+            address: address.clone(),
+        });
+
+        // Since servers handle connection in their own thread, create a channel
+        // to retrieve information
+        let (tx, rx) = std::sync::mpsc::channel();
+        let tx_clone = tx.clone();
+
+        // Start a dummy peer server (although manually, it is the same code as server::listen(),
+        // what's been added is the interaction with the channel)
+        let connection_string = format!("{}:{}", address, peer_port);
+        let connection_string_clone = connection_string.clone();
+        let ipp = inbound_packet_processor.sender().clone();
+        let ipp2 = ipp.clone();
+        let msgr = messenger.sender().clone();
+        let msgr2 = msgr.clone();
+        let ccs = connection_service.sender().clone();
+        let ccs2 = ccs.clone();
+
+        std::thread::spawn(move || {
+            let listener = std::net::TcpListener::bind(connection_string_clone.clone())
+                .expect("Failed to bind socket");
+            trace!("Listening on {:?}", connection_string_clone);
+
+            for stream in listener.incoming() {
+                let stream = stream.expect("Failed to connect to client");
+                let ipp_clone = ipp.clone();
+                let msgr_clone = msgr.clone();
+                let ccs_clone = ccs.clone();
+                let id = uuid::Uuid::new_v4();
+                let sx = tx_clone.clone();
+                thread::spawn(move || {
+                    handle_connection(
+                        stream,
+                        ipp_clone,
+                        msgr_clone,
+                        id,
+                        || ccs_clone.close(id),
+                        sx,
+                    );
+                });
+            }
+        });
+
+        // Start a proper server on its own thread
+        std::thread::spawn(move || {
+            server::listen(ipp2, ccs2, msgr2);
```

Comment:
These variable names are not great. Let's not be stingy with characters and use words

Reply:
Sorry about that, I will rename them appropriately.